galaxy_skeleton1.R

```diff
@@ -7,19 +8,114 @@
 # Read the entire data file into memory using the readLines()-function. Use the
 # URL direcly or read the data from the local file that is in the repository.
 
+raw_data <- readLines("http://www.sao.ru/lv/lvgdb/article/suites_dw_Table1.txt")
+
+
 # Identify the line number L of the separator line between the column names and
 # the rest of the data table.
 
+line_start=substr(x = raw_data, start = 0, stop = 2) #substract first to charakters of each line
+L<- which(line_start=="--") %>% 
+  min() #choose lines that start with "--" and show the first line where that occurs
+L
+
 # Save the variable descriptions (i.e. the information in lines 1:(L-2)) in a
 # text-file for future reference using the cat()-function
 
+var_descriptions<- cat(raw_data[1:(L-2)], sep="\n",file="variable_descriptions.txt")
+
 # Extract the variable names (i.e. line (L-1)), store the names in a vector.
+var_names<- 
+  str_split(string = raw_data[L-1], pattern = "\\|") %>%  #split the row wherever "\\|" occurs
+  unlist() %>% 
+  str_trim()
+var_names
 
 # Read the data. One way to do this is to rewrite the data to a new .csv-file
 # with comma-separators for instance using cat() again, with the variable names
 # from the step above on the first line (see for instance paste() for collapsing
 # that vector to a single string with commas as separators).
 
+comma_separated_values <- 
+  raw_data %>% 
+  gsub("\\|", ",", .) %>% # replace "\\|" with ,
+  gsub(" ", "", .) 
+
+comma_separated_values_with_names <- 
+  c(paste(var_names, collapse = ","), #put together variable names and data
+    comma_separated_values) 
+
+cat(comma_separated_values_with_names, sep = "\n", file = "data_clean.csv") #save in csv file
+
 # Read the finished .csv back into R in the normal way.
+data_clean <-read.csv("data_clean.csv")
+
+
+#PROBLEM 3------
+#Graph which shows that smaller galaxies are underrepresented
+data_clean %>% 
+  slice(15:1000) %>% #exlcuding the first 14 rows with only text data
+  group_by(a_26) %>%
+  ggplot(aes(x=a_26))+
+  geom_bar(fill="blue",width = 0.5) #bar chart shows that small diameters are represented
+#Maybe i didnt get the question, but i dont see how smaller objects are underrepresented here
```

Comment:
If you made a scatterplot with the diameter against the distance to each galaxy you would see that this statement seems to be correct. This is because it is hard to observe smaller objects that are far away than the ones that are close.